### PR TITLE
Use com.yahoo.messagebus.Error instead of java.lang.Error

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli and @jonmv PR

### CODE REVIEW

File: documentapi/src/main/java/com/yahoo/documentapi/Result.java [hunk 1]
```diff
@@ -85,5 +111,16 @@ public enum ResultType {
         @Deprecated(since = "7", forRemoval = true) // TODO: Remove on Vespa 8 — this is a Response outcome, not a Result outcome.
         CONDITION_NOT_MET_ERROR
     }
+    public static Error toError(ResultType result) {
+        switch (result) {
+            case TRANSIENT_ERROR:
+                return new Error(ErrorCode.TRANSIENT_ERROR, ResultType.TRANSIENT_ERROR.name());
+            case CONDITION_NOT_MET_ERROR:
```
Comment (bratseth) [inline documentapi/src/main/java/com/yahoo/documentapi/Result.java:118]: Condition not met isn't transient?

**Review by bratseth — APPROVED** [review]

File: documentapi/src/main/java/com/yahoo/documentapi/Result.java (hunk 1 above)
Comment (baldersheim) [reply documentapi/src/main/java/com/yahoo/documentapi/Result.java:118]: No you will need to change your condition for it to have a bigger chance at success, and anyway it should not be here. See comment in Result.ResultType.

File: documentapi/src/main/java/com/yahoo/documentapi/Result.java [hunk 2]
```diff
@@ -85,5 +111,16 @@ public enum ResultType {
...
+                return new Error(ErrorCode.TRANSIENT_ERROR, ResultType.CONDITION_NOT_MET_ERROR.name());
```
Comment (bratseth) [inline documentapi/src/main/java/com/yahoo/documentapi/Result.java:119]:
```suggestion
                return new Error(ErrorCode.FATAL_ERROR, ResultType.CONDITION_NOT_MET_ERROR.name());
```

**Comment by baldersheim** [comment]
Failure is unrelated

**Review by jonmv — COMMENTED** [review]
:

**Review by vekterli — COMMENTED** [review]
:+1:

**Pull request merged**